### PR TITLE
feat: check ai-app pod healthcheck

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

Check the healthcheck of the AI apps' pods and adds a `health` field to the data sent to the front. 

### What issues does this PR fix or reference?

Part of #265

### CODE REVIEW

File: packages/backend/src/managers/applicationManager.spec.ts [hunk 1]
```diff
@@ -1268,4 +1284,58 @@ describe('getImageTag', () => {
...
     expect(imageTag).equals('quay.io/repo/image:latest');
   });
+
+  test('adoptRunningApplications should check pods health', async () => {
+    mocks.listPodsMock.mockResolvedValue([
+      {
+        Id: 'pod1',
+        engineId: 'engine1',
+        Labels: {
+          'ai-studio-recipe-id': 'recipe-id-1',
```
Comment (feloy) [inline packages/backend/src/managers/applicationManager.spec.ts:1294]: This will need to be changed when #713 is merged

**Comment by axel7083** [comment]
I've noticed pretty recently than podman desktop is sending health events. 

image

I used the same inspect technique with the Inference Manager, and at some point, I am wondering if we should rely on our own pulling update with the inspect mecanism, or wait for podman desktop update maybe @benoitf have an opinion ?

From my understanding, when adding an health check to a container, podman is creating some kind of background services[^1], that will execute the health command on our container, and it seems that this is triggering an event.

I do not think that should be a blocker for this PR, but might be a good way to simplify our code, both in the Inference Manager, and the application manager in the future

[^1]: https://developers.redhat.com/blog/2019/04/18/monitoring-container-vitality-and-availability-with-podman#what_are_healthchecks_

File: packages/backend/src/managers/applicationManager.ts [hunk 2]
```diff
@@ -701,6 +734,39 @@ export class ApplicationManager extends Publisher<ApplicationState[]> {
...
+      );
+      const healthyPod = !containerStates.find(
+        s => s.Status !== 'running' || (s.Health.Status !== '' && s.Health.Status !== 'healthy'),
+      );
+      const recipeId = pod.Labels[LABEL_RECIPE_ID];
+      const modelId = pod.Labels[LABEL_MODEL_ID];
+      if (!this.#applications.has({ recipeId, modelId })) {
+        // a fresh pod could not have been added yet, we will handle it at next iteration
+        continue;
+      }
```
Comment (lstocchi) [inline packages/backend/src/managers/applicationManager.ts:756]: Should this be moved before inspecting all containers?

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**